project-koku/koku
# [COST-3561] Correct path to OCI archived data

## Jira Ticket

[COST-3561](https://issues.redhat.com/browse/COST-3561)

## Description

This change will fix paths to OCI archived data to ensure OCI hive tables are cleared on source deletion

## Testing

1. Checkout this Branch
2. load OCI test data with `make create-test-customer` and `make load-test-customer-data test_source=oci` and wait for ingestion to complete.
3. In your trino shell, check to see that there is data in the OCI hive tables

    ```
    trino:org1234567> select count(*) from oci_cost_line_items;
    _col0 
    -------
    5280 
    (1 row)


    trino:org1234567> select count(*) from oci_cost_line_items_daily;
    _col0 
    -------
    5018 
    (1 row)


    trino:org1234567> select count(*) from oci_usage_line_items;
    _col0 
    -------
    5280 
    (1 row)


    trino:org1234567> select count(*) from oci_usage_line_items_daily;
    _col0 
    -------
    5078 
    (1 row)

    ```

4. Hit this endpoint to delete the OCI source,  http://127.0.0.1:8000/api/cost-management/v1/sources/1/

    - You should see logs similar to this
    ```
    koku-koku-worker-1  | [2023-03-28 17:50:06,347] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Attempting to delete our archived data in S3 under data/parquet/org1234567/OCI/cost/source=d0f7ae55-a246-433e-8ae4-560bc7dd06a5/
    koku-koku-worker-1  | [2023-03-28 17:50:06,514] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Starting objects: 2
    koku-koku-worker-1  | [2023-03-28 17:50:06,550] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Attempting to delete our archived data in S3 under data/parquet/daily/org1234567/OCI/cost/source=d0f7ae55-a246-433e-8ae4-560bc7dd06a5/
    koku-koku-worker-1  | [2023-03-28 17:50:06,668] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Starting objects: 2
    koku-koku-worker-1  | [2023-03-28 17:50:06,701] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Attempting to delete our archived data in S3 under data/parquet/org1234567/OCI/usage/source=d0f7ae55-a246-433e-8ae4-560bc7dd06a5/
    koku-koku-worker-1  | [2023-03-28 17:50:07,039] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Starting objects: 2
    koku-koku-worker-1  | [2023-03-28 17:50:07,073] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Attempting to delete our archived data in S3 under data/parquet/daily/org1234567/OCI/usage/source=d0f7ae55-a246-433e-8ae4-560bc7dd06a5/
    koku-koku-worker-1  | [2023-03-28 17:50:07,184] INFO e267f80b-d7dc-4791-8514-cbf1e6b8d429 Starting objects: 2
    ```

5. And in your trino shell you can check to see that data has been cleared from the OCI hive tables
    
    ```
    trino:org1234567> select count(*) from oci_cost_line_items;
    _col0 
    -------
        0 
    (1 row)

    
    trino:org1234567> select count(*) from oci_cost_line_items_daily;
    _col0 
    -------
        0 
    (1 row)


    trino:org1234567> select count(*) from oci_usage_line_items;
    _col0 
    -------
        0 
    (1 row)
    
    
    trino:org1234567> select count(*) from oci_usage_line_items_daily;
    _col0 
    -------
        0 
    (1 row)

    ```

## Notes

...
